src/aims_ui/models/get_fields.py

```diff
@@ -0,0 +1,178 @@
+        ),
+        common_fields['limit'],
+        common_fields['classification'],
+        common_fields['epoch'],
+        common_fields['england_boost'],
+        common_fields['wales_boost'],
+        common_fields['scotland_boost'],
+        Field(
+            'favourwelsh',
+            search_type='checkbox',
+            display_title='Flavour Welsh',
+            description=
+            'If you prefer the results to be flavoured Welsh tick this box',
+        ),
+        Field(
+            'groupfullpostcodes',
+            search_type='checkbox',
+            display_title='Group full postcodes',
+            description=
+            'If you bwould like the results to be grouped by their respective full postcodes, tick this box',
+        ),
+        Field(
+            'fallback',
+            search_type='checkbox',
+            display_title='Fallback',
+            description=
+            'Specifies whether a slow fallback query is used in the event of the main query returning no results.',
+        ),
+        common_fields['epoch'],
+        common_fields['historical'],
+        common_fields['auxilary_search'],
+    ])
+
+  elif endpoint_name == 'address':
+    pass
+  elif endpoint_name == 'multiple_address':
+    return ([])
+  elif endpoint_name == 'postcode':
+    return ([
+        Field(
+            'postcode',
+            display_title='To get started, enter a PostCode',
+            classes='input--w-8',
+            required=True,
+            description=
+            'A postcode is a code for post which separates houses into groups. This is often a 7 didgit letter and number combo.'
+        ),
+        common_fields['limit'],
+        common_fields['classification'],
+        common_fields['epoch'],
+        common_fields['england_boost'],
+        common_fields['wales_boost'],
+        common_fields['scotland_boost'],
+        common_fields['auxilary_search'],
+    ])
+
+  elif endpoint_name == 'singlesearch':
+    return ([
+        Field(
+            'input',
+            display_title='Enter search string',
+            required=True,
+            description=
+            'Specifies the address search string (e.g. "14 Acacia Avenue, Ruislip, HA4 8RG").'
+        ),
+        common_fields['classification'],
+        common_fields['england_boost'],
+        common_fields['wales_boost'],
+        common_fields['scotland_boost'],
+        common_fields['epoch'],
+        Field(
+            'matchthreshold',
+            display_title='Minimum match %',
+            search_type='dropdown',
+            previous_value='5%',
+            dropdown_options=get_options('percentage_match'),
+            add_default_dropdown_option=False,
+        ),
+        common_fields['limit'],
+        common_fields['historical'],
+        common_fields['auxilary_search'],
+    ])
+
+  elif endpoint_name == 'typeahead':
+    pass
```

Comment:
You might want to add an error-checking else branch here, that errors if the endpoint name didn't match any known endpoints.

Reply:
Done